jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/TaskInstanceServiceImpl.java

```diff
@@ -143,12 +145,66 @@ public long addTask(Task task, ContentData contentData) {
             persistenceContext.persistContent(content);
             persistenceContext.setDocumentToTask(content, contentData, task);
         }
-        
-        
-        taskEventSupport.fireAfterTaskAdded(task, context);
+        fireEvents(task);
         return task.getId();
     }
 
+    private void fireEvents(Task task) {
+        fireWithStatus(task, Status.Created, this::fireAfterTaskAdded);
+        if (task.getTaskData().getStatus() != Status.Created) {
+            fireWithStatus(task, Status.Created, this::fireBeforeTaskActivated);
+            taskEventSupport.fireAfterTaskActivated(task, context);
+        }
+    }
+
+    private void fireWithStatus(Task task, Status eventStatus, Consumer<Task> consumer) {
+        if (task.getTaskData().getStatus() == eventStatus) {
```

Comment:
eventStatus is always Created?

Reply:
You see, after last change we are using fireWithStatus with a different value ;)

Reply:
I see, good point